types/reply.d.ts

```diff
@@ -1,42 +1,52 @@
+  request: FastifyRequest<RawServer, RawRequest, RequestGeneric>;
+  code(statusCode: number): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  status(statusCode: number): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  statusCode: number;
+  sent: boolean;
+  send<T>(payload?: T): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  header(key: string, value: any): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  headers(values: {[key: string]: any}): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
   getHeader(key: string): string | undefined;
   getHeaders(): {
     // Node's `getHeaders()` can return numbers and arrays, so they're included here as possible types.
     [key: string]: number | string | string[] | undefined;
   };
-  // Note: should consider refactoring the argument order for redirect. statusCode is optional so it should be after the required url param
-  redirect(statusCode: number, url: string): FastifyReply<RawServer, RawReply>;
-  redirect(url: string): FastifyReply<RawServer, RawReply>;
   removeHeader(key: string): void;
-  send<T>(payload?: T): FastifyReply<RawServer, RawReply>;
-  serialize(payload: any): string;
-  serializer(fn: (payload: any) => string): FastifyReply<RawServer, RawReply>;
-  status(statusCode: number): FastifyReply<RawServer, RawReply>;
-  statusCode: number;
+  hasHeader(key: string): boolean;
+  // Note: should consider refactoring the argument order for redirect. statusCode is optional so it should be after the required url param
+  redirect(statusCode: number, url: string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  redirect(url: string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  callNotFound(): void;
   getResponseTime(): number;
-  type(contentType: string): FastifyReply<RawServer, RawReply>;
-  context: FastifyContext<ContextConfig>;
-  raw: RawReply;
-  sent: boolean;
+  type(contentType: string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  serializer(fn: (payload: any) => string): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  serialize(payload: any): string;
+  then(fullfilled: () => void, rejected: (err: Error) => void): void;
 }
 
 export type FastifyReply<
   RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
   RawReply extends RawReplyDefaultExpression<RawServer> = RawReplyDefaultExpression<RawServer>,
+  RequestGeneric extends RequestGenericInterface = RequestGenericInterface,
   ContextConfig = ContextConfigDefault
-> = RawReply & FastifyReplyInterface<RawServer, RawReply, ContextConfig>
+> = FastifyReplyInterface<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>
```

Comment:
Looking at this again, I believe it is unnecessary to have this type any longer. The point of breaking them apart was in order to support both declaration merging and generic inheritance. Now that the FastifyRequest and FastifyReply  types are no longer extending from Node.js, but only using it in the `raw` field, I say we rename some things.

I think the best path is to remove this type (and FastifyRequest as well), and then rename the respective interfaces to `FastifyReply` and `FastifyRequest`.

Reply:
+1 on just using `FastifyReply` and `FastifyRequest` idea

Reply:
Done

Reply:
great! I'll check it tomorrow